File: Connascence/Connascence/ConnascenceOfType.py
class ScheduleItem:
    def __init__(self):
        self.flight = ""
        self.origin = ""
        self.destination = ""
        self.departure_dt = None
        self.arrival_dt = None

    def print(self):
        return (
            f"Flight {self.flight} from {self.origin} on {self.departure_dt} "
            f"to {self.destination} on {self.arrival_dt}"
        )


class Itinerary:
    def __init__(self):
        self._schedule_items = []

    def print(self):
        printed_schedules = []
        for item in self._schedule_items:
            # No error, item is added to a collection that is supposed
            # to be a collection of strings. When code runs user will see that
            # itinerary is not printed correctly.
            printed_schedules.append(item)

            # A fix is simple, developer just forgot to call print() on an item.
            # However the error was discovered after the code was run.
            printed_schedules.append(item.print())

        return "\n".join(printed_schedules)

File: Connascence/Connascence/ConnascenceOfType.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ConnascenceOfTYpe
{
    public class ScheduleItem
    {
        public string Flight { get; set; }
        public string Origin { get; set; }
        public string Destination { get; set; }
        public DateTimeOffset DepartureDt { get; set; }
        public DateTimeOffset ArrivalDt { get; set; }

        public string Print()
        {
            return $"Flight {Flight} from {Origin} on {DepartureDt} to {Destination} on {ArrivalDt}";
        }
    }

    public class Itinerary
    {
        private readonly IList<ScheduleItem> _scheduleItems;

        public string Print()
        {
            List<string> printedSchedules = new List<string>();
            foreach (var item in _scheduleItems)
            {
                // COMPILE ERROR
                // printedSchedules is a collection of strings, therefore we can not
                // add ScheduleItem instance into the collection.
                printedSchedules.Add(item);

                // FIXED
                // A fix is simple, developer just forgot to call Print() on item.
                // Print() returns a string that can be added to printedSchedules.
                printedSchedules.Add(item.Print());
            }

            return string.Join(Environment.NewLine, printedSchedules);
        }
    }
}